galsim/chromatic.py

```diff
@@ -227,6 +460,152 @@ def drawImage(self, bandpass, image=None, integrator=None, **kwargs):
+            # Store the weight factors for the two stored images that can contribute at this
+            # wavelength.  Must include the dwave that is part of doing the integral.
+            if hasattr(self, 'SED'):
+                b = self.SED(w)*bandpass(w)*dw[idx]
+            else:
+                b = bandpass(w)*dw[idx]
+            weight_fac[lower_idx] += (1.0-frac)*b
+            weight_fac[lower_idx+1] += frac*b
+
+        # Do the integral as a weighted sum.
+        integral = sum([w*im for w,im in zip(weight_fac, self.ims)])
+
+        # Figure out stepK and maxK using the minimum and maximum (respectively) that have nonzero
+        # weight.  This is the most conservative possible choice, since it's possible that some of
+        # the images that have non-zero weights might have such tiny weights that they don't change
+        # the effective stepk and maxk we should use.
+        stepk = min(np.array(self.stepK_vals)[weight_fac>0])
+        maxk = max(np.array(self.maxK_vals)[weight_fac>0])
+
+        # Instantiate the InterpolatedImage, using these conservative stepK and maxK choices.
+        int_im = galsim.InterpolatedImage(integral, _force_stepk=stepk, _force_maxk=maxk)
+        # Get the transformations at bandpass.red_limit (they are achromatic so it doesn't matter
+        # where you get them).
+        if hasattr(self, '_A'):
+            A0 = self._A(bandpass.red_limit)
+            f0 = self._fluxFactor(bandpass.red_limit)
+            int_im = int_im.transform(A0[0,0], A0[0,1], A0[1,0], A0[1,1])
+            int_im = int_im.shift(A0[0,2], A0[1,2])
+            int_im *= f0
+
+        # Apply integral to the initial image appropriately.  This will naturally work properly and
+        # take into account the supplied value of `add_to_image`, which will be included in kwargs.
+        image = int_im.drawImage(image=image, **kwargs)
+        return image
+
+    def _getCombinedWaveList(self, bandpass):
+        wave_list = bandpass.wave_list
+        wave_list = np.union1d(wave_list, self.wave_list)
+        wave_list = wave_list[wave_list <= bandpass.red_limit]
+        wave_list = wave_list[wave_list >= bandpass.blue_limit]
+        return wave_list
+
+    def _nullTransformation(self, wave):
+        """
+        A utility to check whether the internally stored transformation matrix is consistent with a
+        null transformation at a given wavelength.
+
+        @param wave     Wavelength in nanometers.
+        @returns True/False (True if the transformation corresponds to no shear, shift dilation,
+        or flux rescaling)
+        """
+        null_A = np.matrix(np.identity(3), dtype=float)
+        A0 = self._A(wave)
+        f0 = self._fluxFactor(wave)
+        # Check whether any transformation was done.
+        return not (np.any(abs(A0-null_A) > 1000.*np.finfo(A0.dtype.type).eps) or \
+                        abs(f0-1.) > 1000.*np.finfo(A0.dtype.type).eps)
```

Comment:
1000 is actually a pretty huge slop here.  10 would be plenty.  (Even 1 would probably work.)  It's not such a big deal if dtype is float, but if it is float32, then eps is only 1.e-7, so 1000 * eps is 1.e-4 which is starting to get into the realm of things the user might actually care about.